tensorflow/core/common_runtime/eager/kernel_and_device.cc

```diff
@@ -281,6 +283,49 @@ Status KernelAndDeviceOp::Run(
 
   OpKernelContext context(&params);
 
+  string msg;
+  nvtxRangeId_t nvtx_range;
+
+  if (! nvtx::IsNvtxRangesDisabled()) {
+    if (nvtx::IsNvtxRangesDetailedEnabled()) {
+      std::vector<string> args_pieces;
```

Comment:
I still feel this is too much code to add here. Ideally it should be a single line. So you could create a higher-level function in nvtx namespace that takes inputs as a parameter. You could have an inline definition in the header file that performs the checks and delegates to another function to do the heavy-lifting.